lib/hydra/works/services/characterization_service.rb

```diff
@@ -63,9 +57,10 @@ def extract_metadata(content)
         end
       end
 
+      # Generate a random file name
       def temp_file_name
-        m = %r{/([^/]*)$} .match(object.uri)
-        "#{m[1]}-content.tmp"
+        m = (0...8).map { (65 + rand(26)).chr }.join
```

Comment:
Can we just use the builtin Ruby tempfile function here?  Or is there a reason we need custom tempfile name generation?